lib/Cmts.ml

```diff
@@ -46,14 +46,16 @@ let find_at_position t loc pos =
     [|] character and the first location begins a line and the start column
     of the first location is lower than that of the second location. *)
 let is_adjacent src (l1 : Location.t) (l2 : Location.t) =
-  Option.value_map (Source.string_between src l1.loc_end l2.loc_start)
-    ~default:false ~f:(fun btw ->
-      match String.strip btw with
-      | "" -> true
-      | "|" ->
-          Source.begins_line src l1
-          && Position.column l1.loc_start < Position.column l2.loc_start
-      | _ -> false )
+  match
+    Source.tokens_between src l1.loc_end l2.loc_start ~filter:(function
+      | Token_latest.(COMMENT _ | DOCSTRING _) -> false
```

Comment:
do we need to filter out comments ?

Reply:
Yes otherwise some comments are lost as they are not considered to be adjacent to some term because there is docstrings or other comments in the way (it fails while formatting on lib/Prec.ml)

Reply:
There is something weird here. This is not what the previous implementation was doing.

Reply:
I don't see any failure when i don't filter out comments ...

Reply:
It's weird indeed, I still had the errors until I tried with a fresh new switch, I didn't find where it was coming from, I removed the filter of comments/docstrings then.